cmplibrary/src/main/java/com/sourcepoint/cmplibrary/consent/ClientEventManager.kt

```diff
@@ -91,6 +92,7 @@ private class ClientEventManagerImpl(
             val spConsentString = spConsent
                 ?.let {
                     spClient.onSpFinished(it)
+                    (spClient as? UnitySpClient)?.onSpFinished(it.toJsonObject().toString())
```

Comment:
So we can always call both `onSpFinished(Object)` and `onSpFinished(String)`, without trying to cast the `spClient` into `UnitySpClient`.

In other words, we're adding support to Unity (and potentially other platforms that deal only with strings) right into our `SpClient` rather than as a separate entity.

Reply:
@andresilveirah I'm not that aware of how things work on the Unity side, but if we merge `UnitySpClient` to `SpClient` (which is the main client for Android SDK users) doesn't that mean that Unity SDK will have to override methods with native Kotlin data classes, such as `View`, `SpConsents`, etc?

Reply:
I would rather keep them separate.